src/lib/support/PersistedCounter.h

```diff
@@ -206,12 +206,29 @@ class PersistedCounter : public MonotonicallyIncreasingCounter<T>
         aStartValue = Encoding::LittleEndian::HostSwap<T>(valueLE);
 
 #if CHIP_CONFIG_PERSISTED_COUNTER_DEBUG_LOGGING
-        ChipLogDetail(EventLogging, "PersistedCounter::ReadStartValue() aStartValue 0x%x", aStartValue);
+        // Compiler should optimize these branches.
+        if (std::is_same_v<decltype(aStartValue), uint64_t>)
+        {
+            ChipLogDetail(EventLogging, "PersistedCounter::ReadStartValue() aStartValue 0x" ChipLogFormatX64,
+                          ChipLogValueX64(aStartValue));
+        }
+        else
+        {
+            ChipLogDetail(EventLogging, "PersistedCounter::ReadStartValue() aStartValue 0x%" PRIx32,
+                          static_cast<uint32_t>(aStartValue));
+        }
 #endif
 
         return CHIP_NO_ERROR;
     }
 
+    /**
+     * @brief Get the Initial Counter Value
+     *
+     * @return T PersistedCounter always has 0 as an intial value
```

Comment:
Clearly not always, since subclasses can override....  how about:
```suggestion
     * By default, persisted counters start off at 0.
```